### ISSUE TITLE
`prisma -v` crashes when using `PRISMA_QUERY_ENGINE_BINARY` in specific situation

### ISSUE THREAD

**Issue by janpio** [prisma/prisma#7777]
TLDR: When your project is in a state where you have a download cache for a query engine, but no query engine in your `node_modules/@prisma/engines` because you use `PRISMA_QUERY_ENGINE_BINARY`, it can crash Prisma as it is trying to hash that non existing file anyway.

This is what it looks like:

```
λ npx prisma -v
Error: ENOENT: no such file or directory, open 'C:\Users\Jan\Documents\throwaway\PRISMA_QUERY_ENGINE_BINARY3\node_modules\@prisma\engines\query-engine-windows.exe'
```

```
λ yarn prisma -v
yarn run v1.22.0
$ C:\Users\Jan\Documents\throwaway\PRISMA_QUERY_ENGINE_BINARY\node_modules\.bin\prisma -v
  prisma:loadEnv project root found at C:\Users\Jan\Documents\throwaway\PRISMA_QUERY_ENGINE_BINARY\package.json +0ms
  prisma:tryLoadEnv Environment variables not found at null +0ms
  prisma:tryLoadEnv Environment variables not found at undefined +1ms
  prisma:tryLoadEnv No Environment variables loaded +1ms
  prisma:engines using NAPI: false +0ms
  prisma:engines binaries to download query-engine, migration-engine, introspection-engine, prisma-fmt +1ms
  prisma:download Using env var PRISMA_QUERY_ENGINE_BINARY for binary query-engine, which points to custom-engines/query-engine-windows.exe +0ms
Error: ENOENT: no such file or directory, open 'C:\Users\Jan\Documents\throwaway\PRISMA_QUERY_ENGINE_BINARY\node_modules\@prisma\engines\query-engine-windows.exe'
  prisma:cli Error: ENOENT: no such file or directory, open 'C:\Users\Jan\Documents\throwaway\PRISMA_QUERY_ENGINE_BINARY\node_modules\@prisma\engines\query-engine-windows.exe' +0ms
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

The project is in a situation where we 1. use `PRISMA_QUERY_ENGINE_BINARY` to point to an existing query engine in the project, 2. the cache folder where the CLI stores the binaries (`node_modules/.cache/prisma` in the case of Windows) includes all 4 binaries and their files, 3. `node_modules/@prisma/engines` does only include 3 engines, not the query engine.

```
λ ls node_modules\.cache\prisma\master\c838e79f39885bc8e1611849b1eb28b5bb5bc922\windows\
introspection-engine*           introspection-engine.sha256  migration-engine.gz.sha256  prisma-fmt*           prisma-fmt.sha256  query-engine.gz.sha256
introspection-engine.gz.sha256  migration-engine*            migration-engine.sha256     prisma-fmt.gz.sha256  query-engine*      query-engine.sha256
```

```
λ ls node_modules\@prisma\engines
dist/  download/  introspection-engine-windows.exe*  migration-engine-windows.exe*  package.json  prisma-fmt-windows.exe*  README.md
```

This leads to `prisma -v` triggering some download logic, which along the way tries to hash a query engine binary that does not exist. I think it is missing a check if we are using `PRISMA_QUERY_ENGINE_BINARY`, which is available in that context as `envVarPath` probably already.

### Reproduction

0. Set `DEBUG=*`
1. Create folder, go into it, `npm init -y`, `npm install -D prisma`, `npm install @prisma/client`
2. `npx prisma -v` - should show all all 4 Engines from `node_modules/@prisma/engines`
3. Delete `node_modules/@prisma/engines`: `rm -r node_modules/@prisma/engines`
4. Set `PRISMA_QUERY_ENGINE_BINARY` to existing file (does not _actually_ have to be a query engine file - `package.json` also works)
5. `npm install`
5. Run `npx prisma -v` again
6. Observe the crash above

PS: Yep, this is quite crazy but I have had this and similar things forever and finally found a reproduction via a test that was putting the project in exactly this situation. (Could be worked around by also emptying the `./node_modules/.cache/prisma` folder before running the test)

**Comment by janpio** [comment]
(On Linux, the path to clean for a workaround is `path.join(os.homedir(), '.cache/prisma')`)

**Comment by jkomyno** [comment]
As of `prisma@4.1.0`, this issue is solved thanks to https://github.com/prisma/prisma/pull/14053.
